apps/site/assets/ts/hooks/useVehiclesChannel.ts

```diff
@@ -0,0 +1,137 @@
+import { pick, keyBy } from "lodash";
+import deepEqual from "fast-deep-equal/react";
+import { Reducer } from "react";
+import { DirectionId } from "../__v3api";
+import useChannel from "./useChannel";
+import { CrowdingType } from "../schedule/components/__schedule";
+
+/**
+ * The format of a vehicle emitted by the VehiclesChannel
+ */
+export interface VehicleData {
+  id: string;
+  route_id: string | null;
+  trip_id: string | null;
+  shape_id: string | null;
+  stop_id: string | null;
+  direction_id: DirectionId;
+  longitude: number;
+  latitude: number;
+  bearing: number;
+  status: string;
+  crowding: CrowdingType | null;
+}
+
+export interface Vehicle {
+  id: string;
+  route_id: string | null;
+  trip_id: string | null;
+  shape_id: string | null;
+  stop_id: string | null;
+  direction_id: DirectionId;
+  longitude: number;
+  latitude: number;
+  bearing: number;
+  status: string;
+  crowding: CrowdingType | null;
+}
+
+// Parses departure time into Date()
+export const parseVehicle = (vehicle: VehicleData): Vehicle =>
```

Comment:
This function doesn't parse any dates, comment should be updated.  Also not sure its needed, based on `Vehicle` and `VehicleData` appearing the same.